multimodal/src/autogluon/multimodal/models/ft_transformer.py

```diff
@@ -594,12 +600,23 @@ def __init__(
             initialization="uniform",
         )
 
-        # init weights
+        # init tokenizer and head weights
         if self.numerical_feature_tokenizer:
             self.numerical_adapter.apply(init_weights)
         if self.categorical_feature_tokenizer:
             self.categorical_adapter.apply(init_weights)
         self.head.apply(init_weights)
+        # init transformer backbone from provided checkpoint
+        if checkpoint_name:
+            if "https://" in checkpoint_name or is_s3_url(checkpoint_name):
+                with tempfile.TemporaryDirectory() as tmpdirname:
+                    checkpoint_path = os.path.join(tmpdirname, "./ft_transformer_pretrained.ckpt")
+                    download(checkpoint_name, checkpoint_path)
+                    ckpt = torch.load(checkpoint_path)
+            else:
+                ckpt = torch.load(checkpoint_name)
+            self.transformer.load_state_dict(ckpt["state_dict"])
```

Comment:
This is called every time the model is loaded from disk, even after finetuning. We should ensure we are not using the pretrained weights in any way after we do training, as the pretrained weights are no longer necessary.

Minimal reproducible example:

```python3
from autogluon.tabular import TabularPredictor, TabularDataset


if __name__ == '__main__':
    label = 'class'
    train_data = TabularDataset('https://autogluon.s3.amazonaws.com/datasets/Inc/train.csv')
    test_data = TabularDataset('https://autogluon.s3.amazonaws.com/datasets/Inc/test.csv')
    subsample_size = 500  # subsample subset of data for faster demo, try setting this to much larger values
    if subsample_size is not None and subsample_size < len(train_data):
        train_data = train_data.sample(n=subsample_size, random_state=0)

    hyperparameters = {
        'FT_TRANSFORMER': [
            {"model.ft_transformer.checkpoint_name": "https://automl-mm-bench.s3.amazonaws.com/ft_transformer_pretrained_ckpt/iter_2k.ckpt"},
        ],
    }
    predictor = TabularPredictor(
        label=label,
        eval_metric='roc_auc',
    )

    predictor: TabularPredictor = predictor.fit(train_data, hyperparameters=hyperparameters,)

    # predictor.leaderboard(data=test_data, display=True)

    print("###### Prior to predict  ######")
    predictor.predict(test_data)
    print("###### After Predict (1) ######")
    predictor.predict(test_data)
    print("###### After Predict (2) ######")
    predictor.predict(test_data)
    print("###### After Predict (3) ######")
```

Output:
```
TabularPredictor saved. To load, use: predictor = TabularPredictor.load("AutogluonModels/ag-20240122_233921")
###### Prior to predict  ######
Downloading /tmp/tmplmmfeb91/./ft_transformer_pretrained.ckpt from https://automl-mm-bench.s3.amazonaws.com/ft_transformer_pretrained_ckpt/iter_2k.ckpt...
100%|██████████| 3.13M/3.13M [00:00<00:00, 5.01MiB/s]
Load pretrained checkpoint: /home/ubuntu/workspace/code/scratch/AutogluonModels/ag-20240122_233921/models/FTTransformer/automm_model/model.ckpt
Predicting DataLoader 0: 100%|██████████| 20/20 [00:00<00:00, 47.68it/s]
###### After Predict (1) ######
Downloading /tmp/tmpf5zqen3i/./ft_transformer_pretrained.ckpt from https://automl-mm-bench.s3.amazonaws.com/ft_transformer_pretrained_ckpt/iter_2k.ckpt...
100%|██████████| 3.13M/3.13M [00:00<00:00, 6.26MiB/s]
Load pretrained checkpoint: /home/ubuntu/workspace/code/scratch/AutogluonModels/ag-20240122_233921/models/FTTransformer/automm_model/model.ckpt
Predicting DataLoader 0: 100%|██████████| 20/20 [00:00<00:00, 30.74it/s]
###### After Predict (2) ######
Downloading /tmp/tmp2479loia/./ft_transformer_pretrained.ckpt from https://automl-mm-bench.s3.amazonaws.com/ft_transformer_pretrained_ckpt/iter_2k.ckpt...
100%|██████████| 3.13M/3.13M [00:00<00:00, 4.46MiB/s]
Load pretrained checkpoint: /home/ubuntu/workspace/code/scratch/AutogluonModels/ag-20240122_233921/models/FTTransformer/automm_model/model.ckpt
Predicting DataLoader 0: 100%|██████████| 20/20 [00:00<00:00, 46.99it/s]
###### After Predict (3) ######
```